packages/fx-core/src/plugins/resource/frontend/dotnet/resources/templateInfo.ts

```diff
@@ -24,7 +19,33 @@ export class TemplateInfo {
     this.scenario = TemplateInfo.DefaultScenario;
   }
 
+  static readonly SupportCapability = [Capability.tab, Capability.bot];
   static readonly DonetLanguage = "csharp";
   static readonly DefaultScenario = "default";
+  static readonly BaseGroup = "blazor-base";
   static readonly version = templatesVersion;
 }
+
+export function generateTemplateInfos(selectedCapabilities: string[], ctx: PluginContext) {
+  const projectName = ctx.projectSettings!.appName;
+  const templateVariable: TemplateVariable = { BlazorAppServer: projectName };
+  const templateInfoList: TemplateInfo[] = [];
+
+  selectedCapabilities.forEach((capability) => {
+    if (capability in TemplateInfo.SupportCapability) {
+      templateInfoList.push(new TemplateInfo(ctx, capability.toLowerCase(), templateVariable));
+    }
+  });
+
+  // Generate templateInfo for base scenrio.
+  if (selectedCapabilities.includes(Capability.tab)) {
```

Comment:
Just curious why we use "in" in L35, but use includes() here, given both SupportCapability and selectedCapabilities are List?

Reply:
Fixed. I changed all the code from 'in' to 'include' to make them align. Since 'include' is the typescript style.

Reply:
"in" is not correct here. Actually Array.includes()checks for the existence of a value in an array, while the in operator checks for the existence of a key in an object (or index in the case of arrays). Please check [this link](https://stackoverflow.com/questions/48499181/what-is-the-difference-between-in-operator-and-includes-for-javascript-arr) for more details.
This could be validated by unit test case, so please add some unit test after this PR before we releasing it.

Reply:
Sure!